mdal/frmts/mdal_2dm.cpp

```diff
@@ -286,3 +286,63 @@ std::unique_ptr<MDAL::Mesh> MDAL::Driver2dm::load( const std::string &meshFile,
 
   return std::unique_ptr<Mesh>( mesh.release() );
 }
+
+void MDAL::Driver2dm::save( const std::string &uri, MDAL_Status *status, MDAL::Mesh *mesh )
+{
+  if ( status ) *status = MDAL_Status::None;
+
+  std::ofstream file( uri, std::ofstream::out );
+
+  if ( !file.is_open() )
+  {
+    if ( status ) *status = MDAL_Status::Err_FailToWriteToDisk;
+  }
+
+  std::string line = "MESH2D";
+  file << line << std::endl;
+
+  //write vertices
+  std::unique_ptr<MDAL::MeshVertexIterator> vertexIterator = mesh->readVertices();
+  double vertex[3];
+  for ( size_t i = 0; i < mesh->verticesCount(); ++i )
+  {
+    vertexIterator->next( 1, vertex );
+    line = "ND ";
+    line.append( std::to_string( i + 1 ) );
+    for ( size_t j = 0; j < 3; ++j )
+    {
+      line.append( " " );
+      line.append( std::to_string( vertex[j] ) );
```

Comment:
since it is double, we may not write all digits, but only first few decimal places ... maybe add function to   `bool toBool( const std::string &str );` from mdal_utils.hpp?

Reply:
I am not sure to understand the link between double digits and `bool`.

Reply:
You are right to tell me to be watchful with transform float to string.
This `std::to_string(double)` has only a precision with only 6 digits after the digital point ... With projection coordinates it is ok, but with long/lat, it is a problem because longitudinal length of 1e-5° is equivalent to 1 m when latitude is 0°. So only 6 digits after the digital point is not enough...

But with projected coordinates, more than 2 or 3 numbers after digital point is too much...

In the other hand, for Z value or datasets value, I think it is more pertinent to consider significant digits.

So I purpose to introduce two utils function to handle precision.

Reply:
https://xkcd.com/2170/